src/models/VertexCovering.jl

```diff
@@ -0,0 +1,55 @@
+"""
+$TYPEDEF
+
+Vertex covering is a problem that seeks to find a minimum set of vertices that cover all edges in a graph.
+
+Positional arguments
+-------------------------------
+* `graph` is a graph object.
+* `weights` are associated with the edges of the `graph`, default to `UnitWeight(nv(graph))`.
+"""
+struct VertexCovering{WT<:AbstractVector} <: AbstractProblem
+    graph::SimpleGraph{Int64}
+    weights::WT
+    function VertexCovering(graph::SimpleGraph{Int64}, weights::AbstractVector=UnitWeight(ne(graph)))
+        @assert length(weights) == ne(graph) "length of weights must be equal to the number of edges $(ne(graph)), got: $(length(weights))"
+        new{typeof(weights)}(graph, weights)
+    end
+end
+Base.:(==)(a::VertexCovering, b::VertexCovering) = a.graph == b.graph && a.weights == b.weights
+
+# variables interface
+variables(gp::VertexCovering) = collect(1:nv(gp.graph))
```

Comment:
Each (hyper)edge is associated with a boolean variable, whether this edge is choosen or not.